hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergedReadHandle.java

```diff
@@ -166,6 +167,17 @@ private List<HoodieRecord<T>> doMergedRead(Option<HoodieFileReader> baseFileRead
         String key = record.getRecordKey();
         if (deltaRecordMap.containsKey(key)) {
           deltaRecordKeys.remove(key);
+
+          // When we find a processing time based delete, e.g., a delete record without valid ordering value,
+          // or with default ordering value, i.e., 0, we can safely ignore records before the delete record.
+          // That means, we can return the log record without merging with the record from base file.
+          if (deltaRecordMap.get(key).getMetaDataInfo(DELETE_FOUND_WITHOUT_ORDERING_VALUE).isPresent()) {
+            if (!deltaRecordMap.get(key).isDelete(readerSchema, config.getProps())) {
+              mergedRecords.add(deltaRecordMap.get(key));
+            }
+            continue;
+          }
```

Comment:
Similar here: could `recordMerger` handle this directly?

Reply:
I think in the end, we could incorporate this behavior in the default merger logic. But I am not sure how complex it will be, and how many corner cases.